src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int GROW_FACTOR = 2;
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private List<Entry<K, V>>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new ArrayList[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if ((double) size / buckets.length > LOAD_FACTOR) {
+            resize();
+        }
```

Comment:
Move if block inside resize method and clear from here

Reply:
Ok